code/go/0chain.net/blobbercore/filestore/storage.go

```diff
@@ -0,0 +1,628 @@
+		if err != nil {
+			return nil, nil, common.NewError("minio_download_failed", "Unable to download from minio with err "+err.Error())
+		}
+		file, err = os.Open(fileObjectPath)
+		if err != nil {
+			return nil, nil, err
+		}
+	case err != nil:
+		return nil, nil, err
+	}
+	defer file.Close()
+
+	var returnBytes []byte
+
+	fi, err := file.Stat()
+	if err != nil {
+		return nil, nil, common.NewError("stat_error", err.Error())
+	}
+
+	totalChunks := int(math.Ceil(float64(fi.Size()) / float64(fileData.ChunkSize)))
+
+	fixedMT := util.NewFixedMerkleTree(int(fileData.ChunkSize))
+
+	bytesBuf := bytes.NewBuffer(make([]byte, 0))
+	for i := 0; i < totalChunks; i++ {
+		n, err := io.CopyN(bytesBuf, file, fileData.ChunkSize)
+		if err != nil {
+			if n == 0 && errors.Is(err, io.EOF) {
+				break
+			}
+			return nil, nil, err
+		}
+
+		dataBytes := bytesBuf.Bytes()
+		err = fixedMT.Write(dataBytes, i)
+		if err != nil {
+			return nil, nil, common.NewError("buffer_write_error", err.Error())
+		}
+
+		startInd := blockoffset * MerkleChunkSize
+		endInd := startInd + MerkleChunkSize
+		returnBytes = append(returnBytes, dataBytes[startInd:endInd]...)
+
+		bytesBuf.Reset()
+
+	}
+
+	return returnBytes, fixedMT.GetMerkleTree(), nil
+}
+func (fs FileStore) GetCurrentDiskCapacity() uint64 {
+	return fs.diskCapacity
+}
+
+func (fs FileStore) CalculateCurrentDiskCapacity() error {
+
+	var volStat unix.Statfs_t
+	err := unix.Statfs(fs.mp, &volStat)
+	if err != nil {
+		logging.Logger.Error(fmt.Sprintf("getAvailableSize() unix.Statfs %v", err))
+		return err
+	}
+
+	fs.diskCapacity = volStat.Bavail * uint64(volStat.Bsize)
+	return nil
+}
+
+func (fs *FileStore) isMountPoint() bool {
+	if !filepath.IsAbs(fs.mp) {
+		logging.Logger.Error(fmt.Sprintf("%s is not absolute path", fs.mp))
+		return false
+	}
+
+	/*Below code is temporary fix unless devops comes with exact mountpoint*/
+	if err := os.MkdirAll(fs.mp, 0777); err != nil {
+		logging.Logger.Error(fmt.Sprintf("Error %s while creating directories", err.Error()))
+		return false
+	}
+	if true {
+		return true
+	}
+	/*Above code is temporary fix unless devops comes with exact mountpoint*/
+
+	realMP, err := filepath.EvalSymlinks(fs.mp)
+	if err != nil {
+		logging.Logger.Error(err.Error())
+		return false
+	}
+
+	finfo, err := os.Lstat(realMP)
+	if err != nil {
+		logging.Logger.Error(err.Error())
+		return false
+	}
+
+	pinfo, err := os.Lstat(filepath.Dir(realMP))
+	if err != nil {
+		logging.Logger.Error(err.Error())
+		return false
+	}
+
+	dev := finfo.Sys().(*syscall.Stat_t).Dev
+	pDev := pinfo.Sys().(*syscall.Stat_t).Dev
+
+	return dev != pDev
+}
+
+func (fstr *FileStore) getTemporaryStorageDetails(ctx context.Context, a *allocation, ID string, ch <-chan struct{}, wg *sync.WaitGroup) {
+
+	defer func() {
+		wg.Done()
+		<-ch
+	}()
+
+	var err error
+	defer func() {
+		if err != nil {
+			panic(err)
+		}
+	}()
+
+	tempDir := fstr.getAllocTempDir(ID)
+
+	finfo, err := os.Stat(tempDir)
```

Comment:
How do you mean it? It is the same err variable that is used in defer function.

Reply:
Check the line 483, `finfo, err := os.State(tempDir)`, if we got err here and return in line 488, we expect that the error will be caught in the defer function above, but since the err variable is an new assigned value, hence the err in defer would still be nil, no error will be reported or panic.

Reply:
Never mind, my bad.

Reply:
@peterlimg No it won't create new error variable. It will be the same err. It will only create new `finfo` variable.

Check this out:
```
package main

import (
	"errors"
	"fmt"
)

func main() {
	var err error
	defer func() {
		if err != nil {
			panic(err)
		}
	}()

	i, err := 1, errors.New("hey")
	fmt.Println(i)
}

```